src/prefect/infrastructure/container.py

```diff
@@ -385,6 +386,57 @@ def preview(self):
         finally:
             docker_client.close()
 
+    async def generate_work_pool_base_job_template(self):
+        from prefect.workers.utilities import (
+            get_default_base_job_template_for_infrastructure_type,
+        )
+
+        base_job_template = await get_default_base_job_template_for_infrastructure_type(
+            self.get_corresponding_worker_type()
+        )
+        if base_job_template is None:
+            return await super().generate_work_pool_base_job_template()
+        for key, value in self.dict(exclude_unset=True, exclude_defaults=True).items():
+            if key == "command":
+                base_job_template["variables"]["properties"]["command"]["default"] = (
+                    shlex.join(value)
+                )
+            elif key == "image_registry":
+                self.logger.warning(
+                    "Image registry blocks are not supported by Docker"
+                    " work pools. Please authenticate to your registry using"
+                    " the `docker login` command on your worker instances."
+                )
+            elif key in [
+                "type",
+                "block_type_slug",
+                "_block_document_id",
+                "_block_document_name",
+                "_is_anonymous",
+            ]:
+                continue
+            elif key == "image_pull_policy":
+                new_value = None
+                if value == ImagePullPolicy.ALWAYS:
+                    new_value = "Always"
+                elif value == ImagePullPolicy.NEVER:
+                    new_value = "Never"
+                elif value == ImagePullPolicy.IF_NOT_PRESENT:
+                    new_value = "IfNotPresent"
+
+                base_job_template["variables"]["properties"][key]["default"] = new_value
+            elif key in base_job_template["variables"]["properties"]:
+                base_job_template["variables"]["properties"][key]["default"] = value
+            else:
+                self.logger.warning(
+                    f"Variable {key!r} is not supported by Docker work pools. Skipping."
+                )
+
+        return base_job_template
+
+    def get_corresponding_worker_type(self):
+        return "docker"
```

Comment:
This gets called in `publish_as_work_pool` which `DockerContainer` doesn't override, so it allows customization by the `Infrastructure` subclasses. It might make more sense as a property, but I want each subclass to be able to define what its corresponding work pool type is.